postgresql/provider.go

```diff
@@ -246,6 +255,32 @@ func getRDSAuthToken(region string, profile string, username string, host string
 		return "", err
 	}
 
+	if role != "" {
+		stsClient := sts.NewFromConfig(awscfg)
+		roleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(role),
+			RoleSessionName: aws.String("TerraformSession"),
+		}
+
+		roleOutput, err := stsClient.AssumeRole(ctx, roleInput)
+		if err != nil {
+			return "", err
```

Comment:
I like to wrap all the errors coming from external libs, this helps debugging.
```suggestion
			return "", fmt.Errorf("could not assume AWS role: %w", err)
```